### ISSUE TITLE
LSTM layer after CNN +Glove Embeddings, need to handle false positive/negative sequences scores

### ISSUE THREAD

**Issue by deepankar27** [keras-team/keras#10181]
My Embedding layer:

```
#getting the vocab size & setting the max length of sequences
max_length = 12
t = Tokenizer()
t.fit_on_texts(docs)
vocab_size = len(t.word_index) + 1

#Setting the embedding layer
keras.backend.clear_session()
model = Sequential()
embed = Embedding(vocab_size, 300,weights=[embedding_matrix], input_length=max_length,trainable=False)
model.add(Embedding(vocab_size, 300,weights=[embedding_matrix], input_length=max_length,trainable=False))
model.add(Dropout(0.2))
```

My Model:

```
model.add(Conv1D(vocab_size, 2,padding='same', activation='relu'))
model.add(MaxPooling1D(pool_size=2))
#model.add(LSTM(100,U_regularizer = regularizers.l2(0.01),W_regularizer = regularizers.l2(0.01)))
model.add(LSTM(100,U_regularizer = regularizers.l2(0.001),W_regularizer = regularizers.l2(0.001)))
model.add(Dropout(0.5))
from keras.regularizers import l2
output = model.add(Dense(5, activation='sigmoid'))
optimizer = Adam(lr=0.02, decay=0.02)
#categorical_crossentropy
model.compile(loss='binary_crossentropy', optimizer=optimizer, metrics=['accuracy'])
# summarize the model
print(model.summary())
# fit the model

model.fit(padded_docs, y, epochs=100,batch_size=20, verbose=1)
# evaluate the model
loss, accuracy = model.evaluate(padded_docs, y, verbose=1)
print('Accuracy: %f' % (accuracy*100))
```

Model Summary:
```
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
embedding_2 (Embedding)      (None, 12, 300)           21900     
_________________________________________________________________
dropout_1 (Dropout)          (None, 12, 300)           0         
_________________________________________________________________
conv1d_1 (Conv1D)            (None, 12, 12)            7212      
_________________________________________________________________
max_pooling1d_1 (MaxPooling1 (None, 6, 12)             0         
_________________________________________________________________
lstm_1 (LSTM)                (None, 100)               45200     
_________________________________________________________________
dropout_2 (Dropout)          (None, 100)               0         
_________________________________________________________________
dense_1 (Dense)              (None, 5)                 505       
=================================================================
Total params: 74,817
Trainable params: 52,917
Non-trainable params: 21,900
_________________________________________________________________

```
Learning log:
Epoch 1/100
26/26 [==============================] - 1s 51ms/step - loss: 0.7892 - acc: 0.5077
Epoch 2/100
26/26 [==============================] - 0s 771us/step - loss: 0.5610 - acc: 0.7846
Epoch 3/100
26/26 [==============================] - 0s 719us/step - loss: 0.5927 - acc: 0.8000
Epoch 4/100
26/26 [==============================] - 0s 694us/step - loss: 0.5305 - acc: 0.7769
Epoch 5/100
26/26 [==============================] - 0s 733us/step - loss: 0.4810 - acc: 0.8385
Epoch 6/100
26/26 [==============================] - 0s 761us/step - loss: 0.5221 - acc: 0.8231
Epoch 7/100
26/26 [==============================] - 0s 733us/step - loss: 0.4109 - acc: 0.8846
Epoch 8/100
26/26 [==============================] - 0s 810us/step - loss: 0.3562 - acc: 0.9000
Epoch 9/100
26/26 [==============================] - 0s 771us/step - loss: 0.3441 - acc: 0.8769
Epoch 10/100
26/26 [==============================] - 0s 733us/step - loss: 0.3325 - acc: 0.8769
Epoch 11/100
26/26 [==============================] - 0s 733us/step - loss: 0.3200 - acc: 0.8923
Epoch 12/100
26/26 [==============================] - 0s 771us/step - loss: 0.3517 - acc: 0.9000
Epoch 13/100
26/26 [==============================] - 0s 1ms/step - loss: 0.3245 - acc: 0.8769
Epoch 14/100
26/26 [==============================] - 0s 1ms/step - loss: 0.2413 - acc: 0.9538
Epoch 15/100
26/26 [==============================] - 0s 887us/step - loss: 0.2014 - acc: 0.9538
Epoch 16/100
26/26 [==============================] - 0s 887us/step - loss: 0.2050 - acc: 0.9538
Epoch 17/100
26/26 [==============================] - 0s 1ms/step - loss: 0.1949 - acc: 0.9615
Epoch 18/100
26/26 [==============================] - 0s 926us/step - loss: 0.1667 - acc: 0.9615
Epoch 19/100
26/26 [==============================] - 0s 983us/step - loss: 0.1662 - acc: 0.9769
Epoch 20/100
26/26 [==============================] - 0s 1ms/step - loss: 0.1578 - acc: 0.9769
Epoch 21/100
26/26 [==============================] - 0s 1ms/step - loss: 0.1470 - acc: 0.9923
Epoch 22/100
26/26 [==============================] - 0s 926us/step - loss: 0.1363 - acc: 0.9769
Epoch 23/100
26/26 [==============================] - 0s 887us/step - loss: 0.1287 - acc: 0.9923
Epoch 24/100
26/26 [==============================] - 0s 983us/step - loss: 0.1148 - acc: 1.0000
Epoch 25/100
26/26 [==============================] - 0s 983us/step - loss: 0.1180 - acc: 1.0000
Epoch 26/100
26/26 [==============================] - 0s 868us/step - loss: 0.1078 - acc: 1.0000
Epoch 27/100
26/26 [==============================] - 0s 945us/step - loss: 0.0984 - acc: 1.0000
Epoch 28/100
26/26 [==============================] - 0s 791us/step - loss: 0.0996 - acc: 0.9923
Epoch 29/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0891 - acc: 1.0000
Epoch 30/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0833 - acc: 1.0000
Epoch 31/100
26/26 [==============================] - 0s 983us/step - loss: 0.0839 - acc: 1.0000
Epoch 32/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0799 - acc: 1.0000
Epoch 33/100
26/26 [==============================] - 0s 2ms/step - loss: 0.0715 - acc: 1.0000
Epoch 34/100
26/26 [==============================] - 0s 945us/step - loss: 0.0701 - acc: 1.0000
Epoch 35/100
26/26 [==============================] - 0s 829us/step - loss: 0.0817 - acc: 0.9923
Epoch 36/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0753 - acc: 1.0000
Epoch 37/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0613 - acc: 1.0000
Epoch 38/100
26/26 [==============================] - 0s 964us/step - loss: 0.0601 - acc: 1.0000
Epoch 39/100
26/26 [==============================] - 0s 926us/step - loss: 0.0686 - acc: 1.0000
Epoch 40/100
26/26 [==============================] - 0s 964us/step - loss: 0.0560 - acc: 1.0000
Epoch 41/100
26/26 [==============================] - 0s 887us/step - loss: 0.0551 - acc: 1.0000
Epoch 42/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0505 - acc: 1.0000
Epoch 43/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0566 - acc: 1.0000
Epoch 44/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0494 - acc: 1.0000
Epoch 45/100
26/26 [==============================] - 0s 983us/step - loss: 0.0510 - acc: 1.0000
Epoch 46/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0478 - acc: 1.0000
Epoch 47/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0490 - acc: 1.0000
Epoch 48/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0473 - acc: 1.0000
Epoch 49/100
26/26 [==============================] - 0s 887us/step - loss: 0.0452 - acc: 1.0000
Epoch 50/100
26/26 [==============================] - 0s 964us/step - loss: 0.0463 - acc: 1.0000
Epoch 51/100
26/26 [==============================] - 0s 868us/step - loss: 0.0413 - acc: 1.0000
Epoch 52/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0402 - acc: 1.0000
Epoch 53/100
26/26 [==============================] - 0s 964us/step - loss: 0.0395 - acc: 1.0000
Epoch 54/100
26/26 [==============================] - 0s 887us/step - loss: 0.0401 - acc: 1.0000
Epoch 55/100
26/26 [==============================] - 0s 965us/step - loss: 0.0380 - acc: 1.0000
Epoch 56/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0376 - acc: 1.0000
Epoch 57/100
26/26 [==============================] - 0s 964us/step - loss: 0.0405 - acc: 1.0000
Epoch 58/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0353 - acc: 1.0000
Epoch 59/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0349 - acc: 1.0000
Epoch 60/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0332 - acc: 1.0000
Epoch 61/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0321 - acc: 1.0000
Epoch 62/100
26/26 [==============================] - 0s 926us/step - loss: 0.0339 - acc: 1.0000
Epoch 63/100
26/26 [==============================] - 0s 771us/step - loss: 0.0350 - acc: 1.0000
Epoch 64/100
26/26 [==============================] - 0s 926us/step - loss: 0.0365 - acc: 1.0000
Epoch 65/100
26/26 [==============================] - 0s 906us/step - loss: 0.0317 - acc: 1.0000
Epoch 66/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0316 - acc: 1.0000
Epoch 67/100
26/26 [==============================] - 0s 926us/step - loss: 0.0331 - acc: 1.0000
Epoch 68/100
26/26 [==============================] - 0s 907us/step - loss: 0.0307 - acc: 1.0000
Epoch 69/100
26/26 [==============================] - 0s 996us/step - loss: 0.0319 - acc: 1.0000
Epoch 70/100
26/26 [==============================] - 0s 887us/step - loss: 0.0295 - acc: 1.0000
Epoch 71/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0354 - acc: 1.0000
Epoch 72/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0299 - acc: 1.0000
Epoch 73/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0339 - acc: 1.0000
Epoch 74/100
26/26 [==============================] - 0s 887us/step - loss: 0.0321 - acc: 1.0000
Epoch 75/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0311 - acc: 1.0000
Epoch 76/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0289 - acc: 1.0000
Epoch 77/100
26/26 [==============================] - 0s 848us/step - loss: 0.0302 - acc: 1.0000
Epoch 78/100
26/26 [==============================] - 0s 887us/step - loss: 0.0273 - acc: 1.0000
Epoch 79/100
26/26 [==============================] - 0s 945us/step - loss: 0.0304 - acc: 1.0000
Epoch 80/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0256 - acc: 1.0000
Epoch 81/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0273 - acc: 1.0000
Epoch 82/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0296 - acc: 1.0000
Epoch 83/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0238 - acc: 1.0000
Epoch 84/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0268 - acc: 1.0000
Epoch 85/100
26/26 [==============================] - 0s 713us/step - loss: 0.0280 - acc: 1.0000
Epoch 86/100
26/26 [==============================] - 0s 848us/step - loss: 0.0265 - acc: 1.0000
Epoch 87/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0303 - acc: 1.0000
Epoch 88/100
26/26 [==============================] - 0s 964us/step - loss: 0.0249 - acc: 1.0000
Epoch 89/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0228 - acc: 1.0000
Epoch 90/100
26/26 [==============================] - 0s 733us/step - loss: 0.0285 - acc: 1.0000
Epoch 91/100
26/26 [==============================] - 0s 945us/step - loss: 0.0229 - acc: 1.0000
Epoch 92/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0268 - acc: 1.0000
Epoch 93/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0227 - acc: 1.0000
Epoch 94/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0233 - acc: 1.0000
Epoch 95/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0247 - acc: 1.0000
Epoch 96/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0229 - acc: 1.0000
Epoch 97/100
26/26 [==============================] - 0s 1ms/step - loss: 0.0238 - acc: 1.0000
Epoch 98/100
26/26 [==============================] - 0s 926us/step - loss: 0.0257 - acc: 1.0000
Epoch 99/100
26/26 [==============================] - 0s 887us/step - loss: 0.0233 - acc: 1.0000
Epoch 100/100
26/26 [==============================] - 0s 829us/step - loss: 0.0230 - acc: 1.0000
26/26 [==============================] - 0s 4ms/step
Accuracy: 100.000000

I am using very small set of data, max 50 sentences and 5 labels. Two of them has only two sentences. 
Now, for true positive sequence it gives good score. For more clarity I will share example.

Label 1 has sequence -> My account got locked.
It gives accuracy of 0.945
Next, if I want to classify sequence which is false positive -> My door got locked then it gives me 0.8234
But I expect at lest less than 0.7 score for such sequences.  
I tried to use different loss functions:

```
def cos_distance(y_true, y_pred):
    y_true = K.l2_normalize(y_true, axis=-1)
    y_pred = K.l2_normalize(y_pred, axis=-1)
    return K.mean(1 - K.sum((y_true * y_pred), axis=-1))
```
Even this is giving varied scores. 
Can you guys give a direction how can I we bring a good score gap between true positive & false positive sequences? 
I am open to any views & I am new to this domain so, anyway it will be a good learning for me.

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
